subsys/bluetooth/host/audio/bass.c

```diff
@@ -929,17 +948,26 @@ static int bass_mod_src(struct bt_conn *conn, struct net_buf_simple *buf)
 		}
 	}
 
-	internal_state->pa_interval = pa_interval;
+	if (state->pa_sync != pa_sync) {
+		state->pa_sync = pa_sync;
+		need_synced = true;
+		state_changed = true;
+	}
 
-	/* If no change to the state don't sync PA */
-	if (!state_changed) {
-		bass_pa_sync(conn, internal_state, pa_sync);
+	internal_state->pa_interval = pa_interval;
 
+	if (need_synced) {
```

Comment:
For update metadata only, does we need resync big?

Reply:
Been a while since I looked at this, but I don't think so, no. 

However, the Metadata contain information about the codec, so we might need to change the data path (not something the current ISO supports). 

I would need to check the BAP spec again to see if a broadcast source is allowed to change the codec after the BIG has been created or not. It is, however, a very unlikely situation, so I don't think we need to support that at the moment in any case.

Reply:
As far as I know, metadata does not contain codec information. Codec configuration information is in codec configuration field, which is in PAC Record.

Reply:
You are correct; I remember that Codec was part of the BASE, but not in the metadata field. Then no, we shouldn't need to do any resync or anything based on metadata changes.

Reply:
Audio Capability Discovery
Discovery of a Sink PAC characteristic informs the Broadcast Assistant that the Broadcast Sink collocated with the Scan Delegator is capable of receiving and decoding audio data encoded using the settings defined as Mandatory in Table 3.17.
The Broadcast Assistant may read the value of Sink PAC characteristics to discover audio capability settings not defined as Mandatory in Table 3.17 (for example, audio capabilities defined by higher-layer specifications or vendor-specific audio capabilities defined by an implementation) that are supported by the Broadcast Sink collocated with the Scan Delegator.